### PR TITLE
adding automountServiceAccountToken to helm chart

### PR DESCRIPTION
**Pull request by jasonwilliams14**
### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [ x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] I have checked that all unit tests pass after adding my changes
- [ x] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/values.yaml [hunk 1]
```diff
@@ -96,6 +96,9 @@ controller:
     ## Format: <namespace>/<secret_name>
     secret:
 
+  ## The option to toggle automountServiceToken. 
+  automountServiceToken: true
```
Comment (lucacome) [inline deployments/helm-chart/values.yaml:100]:
```suggestion
  ## The option to toggle automountServiceAccountToken at the pod level. 
  automountServiceAccountToken: true 
```

File: docs/content/installation/installation-with-helm.md [hunk 2]
```diff
@@ -171,6 +171,7 @@ The following tables lists the configurable parameters of the NGINX Ingress Cont
 |``controller.wildcardTLS.cert`` | The base64-encoded TLS certificate for every Ingress/VirtualServer host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress/VirtualServer hosts NGINX will break any attempt to establish a TLS connection. | None |
 |``controller.wildcardTLS.key`` | The base64-encoded TLS key for every Ingress/VirtualServer host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress/VirtualServer hosts NGINX will break any attempt to establish a TLS connection. | None |
 |``controller.wildcardTLS.secret`` | The secret with a TLS certificate and key for every Ingress/VirtualServer host that has TLS enabled but no secret specified. The value must follow the following format: ``<namespace>/<name>``. Used as an alternative to specifying a certificate and key using ``controller.wildcardTLS.cert`` and ``controller.wildcardTLS.key`` parameters. | None |
+``controller.automountServiceAccountToken`` | Setting automountServiceAccountToken at the pod level | true
```
Comment (lucacome) [inline docs/content/installation/installation-with-helm.md:174]:
```suggestion
|``controller.automountServiceAccountToken`` | Setting automountServiceAccountToken at the pod level | true
```

File: deployments/helm-chart/README.md [hunk 3]
```diff
@@ -168,6 +168,7 @@ Parameter | Description | Default
 `controller.wildcardTLS.cert` | The base64-encoded TLS certificate for every Ingress/VirtualServer host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress/VirtualServer hosts NGINX will break any attempt to establish a TLS connection. | None
 `controller.wildcardTLS.key` | The base64-encoded TLS key for every Ingress/VirtualServer host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress/VirtualServer hosts NGINX will break any attempt to establish a TLS connection. | None
 `controller.wildcardTLS.secret` | The secret with a TLS certificate and key for every Ingress/VirtualServer host that has TLS enabled but no secret specified. The value must follow the following format: `<namespace>/<name>`. Used as an alternative to specifying a certificate and key using `controller.wildcardTLS.cert` and `controller.wildcardTLS.key` parameters. | None
+`controller.automountServiceAccountToken` | Setting for automountServiceAccountToken for NGINX Ingress. | true
```
Comment (lucacome) [inline deployments/helm-chart/README.md:171]:
```suggestion
`controller.automountServiceAccountToken` | The toggle for automountServiceAccountToken for the Ingress Controller. | true
```

**Comment by lucacome** [comment]
Not sure if my suggestions for the descriptions make sense, calling in @nginxinc/nginx-docs 😄

File: deployments/helm-chart/values.yaml (hunk 1 above)
Comment (jasonwilliams14) [reply deployments/helm-chart/values.yaml:100]: Typo on my part.

**Comment by jasonwilliams14** [comment]
Closed duplicate PR and added files for the manifest files as needed.

**Review by ciarams87 — APPROVED** [review]

**Pull request merged**